### PR TITLE
upcxx: Update the UPC++ package to 2021.9.0

### PR DESCRIPTION
**Pull request by bonachea**
Changes: 

* Add the new release, and a missing older one.

* Remove the spack package cruft for supporting the obsolete build system that
  was present in older versions that are no longer supported.

* General cleanups.

Support for library versions older than 2020.3.0 is officially retired,
for two reasons:

1. Releases prior to 2020.3.0 had a required dependency on Python 2,
   which is [officially EOL](https://www.python.org/doc/sunset-python-2/)
   as of Jan 1 2020, and is no longer considered secure.
2. (Most importantly) The UPC++ development team is unable/unwilling to
   support releases more than two years old.  UPC++ provides robust
   backwards-compatibility to earlier releases of UPC++ v1.0, with very
   rare well-documented/well-motivated exceptions.  Users are strongly
   encouraged to update to a current version of UPC++.

NOTE: Most of the lines changed in this commit are simply re-indentation,
and thus might be best reviewed in a diff that ignores whitespace.

### CODE REVIEW

**Review by PHHargrove — DISMISSED** [review]
Looks good to me.

**Comment by bernhardkaindl** [comment]
@bonachea This PR is currently a draft and we can't merge it as draft. When you are ready for merge/maintainer-review click the button labelled "Ready for review" above the list of passed checks right of this text:
> This pull request is still a work in progress
Draft pull requests cannot be merged.

**Comment by bonachea** [comment]
> @bonachea This PR is currently a draft and we can't merge it as draft. When you are ready for merge/maintainer-review click the button labelled "Ready for review" above the list of passed checks right of this text:

Thanks @bernhardkaindl - One of our maintainers is on vacation this week, so I've deliberately left this PR in draft mode to keep it "on hold" until he returns, allowing him to be involved in any potential discussion with reviewers.

**Comment by bonachea** [comment]
I've just rebased onto the latest develop and appended a commit that fixes an incompatibility on new Cray Shasta systems.

**Review by PHHargrove — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**
